### PR TITLE
Add package documentation

### LINKED ISSUE #1219: Document the geotrellis.slick package
There needs to be docs for the geotrellis-slick package. Explaining what is needed to run the unit tests should be included.

### PR DESCRIPTION
**Pull request by moradology**
- [x] Document Slick
- [x] Document GDAL
- [x] Document Spark
- [x] Document macros
- [x] Improve contribution docs
- [x] Improve docs root

Fixes #1219

### CODE REVIEW

**Comment by moradology** [comment]
This didn't come quite as far as I'd have liked it to - but the bulk of this work should be complete

File: gdal/src/main/scala/geotrellis/gdal/io/hadoop/NetCdfInputFormat.scala [hunk 1]
```diff
@@ -0,0 +1,136 @@
+package geotrellis.gdal.io.hadoop
```
Comment (lossyrob) [inline gdal/src/main/scala/geotrellis/gdal/io/hadoop/NetCdfInputFormat.scala]: Remove GDAL!

File: spark/src/main/scala/geotrellis/spark/package.scala [hunk 2]
```diff
@@ -70,11 +70,11 @@ package object spark
       new ContextRDD(rdd, metadata)
   }
 
-  type Component[T, C] = PLens[T, T, C, C]
+  type Component[T, C] = Lens[T, C]
 
   object Component {
     def apply[T, C](get: T => C, set: (T, C) => T): Component[T, C] =
-      PLens[T, T, C, C](get)(c => t => set(t, c))
+      Lens[T, C](get)(c => t => set(t, c))
```
Comment (lossyrob) [inline spark/src/main/scala/geotrellis/spark/package.scala]: putting this change in another PR that will make RC2, can remove when you merge in master

**Comment by lossyrob** [comment]
Superseded by #1424

**Pull request closed without merging**